src/goto-symex/goto_symex_state.cpp

```diff
@@ -241,15 +215,19 @@ void goto_symex_statet::assignment(
   // the type might need renaming
   rename(lhs.type(), l1_identifier, ns);
   lhs.update_type();
-  assert_l1_renaming(lhs);
-
-  #if 0
+  const validation_modet local_validation_mode = validation_modet::INVARIANT;
```

Comment:
If the validation mode is hardcoded to `validation_modet::INVARIANT` we could also use `DATA_INVARIANT()` directly instead. I think when using `DATA_CHECK()` the mode should be truly configurable.